internal_transfers/actions/src/parse.ts

```diff
@@ -23,32 +20,42 @@ export function partitionEventLogs(logs: Log[]): ClassifiedEvents {
     trades: [],
     settlements: [],
   };
+  const settlementInterface = new ethers.Interface(SETTLEMENT_ABI);
+  const erc20Interface = new ethers.Interface(IERC20_ABI);
   logs.forEach((log, index) => {
-    const topics = log.topics;
-    const eventTopic = topics[0];
-
-    if (eventTopic === TRANSFER_EVENT_TOPIC) {
+    // We are only interested in Transfer Events from erc20 contracts
+    // along with Settlement and Trade Events from the Settlement contract
+    // All other event emissions can be ignored for our purposes.
+    const transferEventLog = erc20Interface.parseLog(log);
+    const settlementEventLog = settlementInterface.parseLog(log);
+    if (transferEventLog != null) {
+      const { from, to } = transferEventLog.args;
       const transfer: TransferEvent = {
-        to: addressFromBytes32(topics[2]),
-        from: addressFromBytes32(topics[1]),
-        amount: BigInt(log?.data),
+        from,
+        to,
+        amount: BigInt(transferEventLog.args[2]),
       };
       // Is a "relevant" transfer (involving settlement contract)
       if (transferInvolves(transfer, SETTLEMENT_CONTRACT_ADDRESS)) {
         result.transfers.push(transfer);
       }
```

Comment:
Shouldn't this step also check if `settlementEventLog.name == "Transfer"` to avoid including other events like `Approval`?

Reply:
I don't believe a  `settlementEvent` can ever be a transfer. Here we are referring to an event that is already known to be a Transfer. 

However, you are correct to notice that the transferEventLog could have been Approve. I have fixed this in a line above. one line before what you highlighted here.

```
if (transferEventLog != null && transferEventLog.name === "Transfer") {
```